File: BasicRxJavaSample/app/src/main/java/com/example/android/observability/demoTest/ImitateUserDataSource.java
package com.example.android.observability.demoTest;


import com.example.android.observability.demoTest.persisitence.ImitateUser;

import io.reactivex.Completable;
import io.reactivex.Flowable;

public interface ImitateUserDataSource {

    Flowable<ImitateUser> getUser();

    Completable insertOrUpdateUser(ImitateUser imitateUser);

    void deleteAllUsers();
}

File: BasicRxJavaSampleKotlin/app/src/main/java/com/example/android/observability/imitate/persistance/ImitateUserDataSource.kt
package com.example.android.observability.imitate.persistance

import io.reactivex.Completable
import io.reactivex.Flowable

interface ImitateUserDataSource {
    public fun getUser():Flowable<ImitateUser>
    public fun updateOrInsertUser(imitateuser:ImitateUser):Completable
}